Adjust log4j dependencies for new MF release

> Caused by: java.lang.IllegalStateException: Detected both log4j-over-slf4j.jar AND bound slf4j-reload4j.jar on the class path, preempting StackOverflowError. See also http://www.slf4j.org/codes.html#log4jDelegationLoop for more details.
>         at org.lobid.resources.AlmaMarc21XmlToLobidJsonMetafixTest.<clinit>(AlmaMarc21XmlToLobidJsonMetafixTest.java:48)
> 
> https://stackoverflow.com/a/18320480 seems that we have to adjust something, does this mean we have an breaking change? 

 _Originally posted by @TobiasNx in [#713](https://github.com/metafacture/metafacture-core/issues/713#issuecomment-3882736460)_


This should be finished before moving on with the release.

--- a/pom.xml
+++ b/pom.xml
@@ -205,11 +205,6 @@
             <artifactId>logback-core</artifactId>
             <version>${logback.version}</version>
         </dependency>
-        <dependency>
-            <groupId>org.slf4j</groupId>
-            <artifactId>slf4j-reload4j</artifactId>
-            <version>1.7.36</version>
-        </dependency>
     </dependencies>
     <build>
         <resources>

